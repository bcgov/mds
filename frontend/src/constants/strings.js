import React from "react";
// reusable messages for error handling, validations, null-screens, etc.
export const ERROR = "Error!";
export const LOADING = "Loading...";
export const TRY_AGAIN = "Please try again later";
export const NO_DATA = "No data available";
export const NO_COORDINATES = "This mine does not contain valid coordinates";
export const NO_MINE_MANAGER = "No assigned mine manager";
export const NO_PREV_MINE_MANAGER = "No previous mine managers found";
export const NO_PERMITTEE = "No assigned permittee";
export const NO_PERMIT = "No permit information available";
export const NO_APPLICATION = "No application information available";
export const ADD_MINE_MANAGER = "Please add mine manager below";
export const ADD_TENURE = "Please add tenure number below";
export const ADD_PARTY = "Please create party below";
export const NO_RESULTS = "No results found";
export const NO_NRIS_INSPECTIONS = "No inspections in NRIS";
export const MAP_UNAVAILABLE = [
  "The BC Geographic Warehouse (BCGW) map layers used in this application are unavailable at this time.",
  "Please notify ",
  <b>mds@gov.bc.ca</b>,
  " if this problem persists.",
];

export const UNAUTHORIZED = "You do not have permission to access this site";
export const UNAUTHORIZED_PAGE = "You do not have permission to access this page";
// to get HTML to render in a React String you send the string as a list of its parts, a mix of HTML and strings.
export const CONTACT_ADMIN = [
  "Contact your system administrator at ",
  <b>mds@gov.bc.ca</b>,
  " to request access",
];
export const EMPTY_FIELD = "N/A";

// default coordinates for center of BC
export const DEFAULT_LAT = 53.7267;
export const DEFAULT_LONG = -127.6476;

// default url values
export const DEFAULT_PAGE = "1";
export const DEFAULT_PER_PAGE = "25";
export const DEFAULT_DASHBOARD_PARAMS = "?page=1&per_page=25";

// mine types
export const MAJOR_MINE = "Major mine";
export const REGIONAL_MINE = "Regional mine";
