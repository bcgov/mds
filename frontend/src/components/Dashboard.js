/**
 * @class Dasboard is the main landing page of the application, currently containts a list of viewable mines and the ability to add a new mine.
 */
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Avatar, Badge, Button, Col, Card, Row } from 'antd';

import { getMineRecords } from '@/actionCreators/mineActionCreator';
import { getMines, getMineIds } from '@/selectors/mineSelectors';
import * as router from '@/constants/routes';

const propTypes = {
  getMineRecords: PropTypes.func.isRequired,
  mines: PropTypes.object.isRequired,
  mineIds: PropTypes.array.isRequired,
};

const defaultProps = {
  mines: {},
  mineIds: [],
};

class Dashboard extends Component {
  componentDidMount() {
    this.props.getMineRecords();
  }

  render() {
    const { mines, mineIds } = this.props;
    return (
      <div>
        <h1> Mine Dashboard </h1>
        <Card style={{ textAlign: "center" }}>
          <Link to={router.CREATE_MINE_RECORD.route}>
            <div>
              <Badge count={"+"}>
                <Avatar shape="square" size="large" icon="database" />
              </Badge>
            </div>
            <div style={{ padding: "10px" }}>
              <Button type="primary" size="small" >
                Create Mine Record
            </Button>
            </div>
          </Link>
        </Card>
        <Card title="Mines">
          <Row type="flex">
            <Col span={4}><strong>MINE_NO</strong></Col>
            <Col span={8}><strong>NAME</strong></Col>
            <Col span={8}><strong>GUID</strong></Col>
            <Col span={4}><strong>ACTION</strong></Col>
          </Row>
          {mineIds.map((id) => {
            return (
              <div style={{padding: "10px"}} key={id}>
                <Row type="flex">
                  <Col span={4}>{mines[id].mine_detail[0] ? mines[id].mine_detail[0].mine_no : "-"}</Col>
                  <Col span={8}>{mines[id].mine_detail[0] ? mines[id].mine_detail[0].mine_name : "-"}</Col>
                  <Col span={8}>{mines[id].guid}</Col>
                  <Col span={4}>
                    <Link to={router.MINE_SUMMARY.dynamicRoute(mines[id].mine_detail[0] ? mines[id].mine_detail[0].mine_no : "")}>
                      <Button type="primary" size="small" >
                        View
                      </Button>
                    </Link>
                  </Col>
                </Row>
              </div>
            )
          })
          }
        </Card>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    mines: getMines(state),
    mineIds: getMineIds(state),
  };
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    getMineRecords
  }, dispatch);
};

Dashboard.propTypes = propTypes;
Dashboard.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
