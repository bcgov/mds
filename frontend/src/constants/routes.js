import queryString from "query-string";
import Home from "@/components/Home";
import Dashboard from "@/components/dashboard/minesHomePage/Dashboard";
import ContactHomePage from "@/components/dashboard/contactsHomePage/ContactHomePage";
import MineDashboard from "@/components/mine/MineDashboard";
import PartyProfile from "@/components/parties/PartyProfile";
import RelationshipProfile from "@/components/parties/RelationshipProfile";
import AdminDashboard from "@/components/admin/AdminDashboard";

export const DASHBOARD = {
  route: "/",
  component: Home,
};

export const MINE_HOME_PAGE = {
  route: "/dashboard/mines",
  dynamicRoute: ({ page, per_page, ...params }) =>
    `/dashboard/mines/?${queryString.stringify({ page, per_page, ...params }, { sort: false })}`,
  mapRoute: (page, perPage, search = null) => {
    const searchParam = search ? `&search=${search}` : "";
    return `/dashboard/mines?page=${page}&per_page=${perPage}${searchParam}&map=true`;
  },
  component: Dashboard,
};

export const CONTACT_HOME_PAGE = {
  route: "/dashboard/contacts",
  dynamicRoute: ({ page, per_page, ...params }) =>
    `/dashboard/contacts/?${queryString.stringify({ page, per_page, ...params }, { sort: false })}`,
  component: ContactHomePage,
};

export const MINE_SUMMARY = {
  route: "/dashboard/:id/:activeTab",
  dynamicRoute: (id, activeTab = "summary") => `/dashboard/${id}/${activeTab}`,
  component: MineDashboard,
};

export const PARTY_PROFILE = {
  route: "/dashboard/:id/profile",
  dynamicRoute: (id) => `/dashboard/${id}/profile`,
  component: PartyProfile,
};

export const RELATIONSHIP_PROFILE = {
  route: "/dashboard/:id/history/:typeCode",
  dynamicRoute: (id, typeCode) => `/dashboard/${id}/history/${typeCode}`,
  component: RelationshipProfile,
};

export const ADMIN_DASHBOARD = {
  route: "/admin/dashboard",
  component: AdminDashboard,
};
