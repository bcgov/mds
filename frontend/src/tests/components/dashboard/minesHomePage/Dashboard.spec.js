import React from "react";
import { shallow } from "enzyme";
import { Dashboard } from "@/components/dashboard/minesHomePage/Dashboard";
import * as MOCK from "@/tests/mocks/dataMocks";
import * as router from "@/constants/routes";
import * as String from "@/constants/strings";

const dispatchProps = {};
const reducerProps = {};

const setupDispatchProps = () => {
  dispatchProps.fetchMineRecords = jest.fn(() => Promise.resolve({}));
  dispatchProps.createMineRecord = jest.fn();
  dispatchProps.fetchStatusOptions = jest.fn();
  dispatchProps.fetchRegionOptions = jest.fn();
  dispatchProps.fetchMineDisturbanceOptions = jest.fn();
  dispatchProps.fetchMineTenureTypes = jest.fn();
  dispatchProps.fetchMineCommodityOptions = jest.fn();
  dispatchProps.openModal = jest.fn();
  dispatchProps.closeModal = jest.fn();
  dispatchProps.fetchPartyRelationshipTypes = jest.fn();
  dispatchProps.fetchPermitStatusOptions = jest.fn();
  dispatchProps.setOptionsLoaded = jest.fn();
};

const setupReducerProps = () => {
  reducerProps.location = { search: " " };
  reducerProps.history = {
    push: jest.fn(),
    location: {},
  };
  reducerProps.mines = MOCK.MINES.mines;
  reducerProps.mineIds = MOCK.MINES.mineIds;
  reducerProps.pageData = MOCK.PAGE_DATA;
  reducerProps.mineStatusOptions = MOCK.STATUS_OPTIONS.options;
  reducerProps.mineRegionOptions = MOCK.REGION_OPTIONS.options;
  reducerProps.mineDisturbanceOptions = MOCK.DISTURBANCE_OPTIONS;
  reducerProps.mineRegionHash = MOCK.REGION_HASH;
  reducerProps.mineTenureTypes = MOCK.TENURE_TYPES.options;
  reducerProps.mineTenureHash = MOCK.TENURE_HASH;
};

beforeEach(() => {
  setupDispatchProps();
  setupReducerProps();
});

describe("Dashboard", () => {
  it("renders properly", () => {
    const component = shallow(<Dashboard {...dispatchProps} {...reducerProps} />);
    expect(component).toMatchSnapshot();
  });

  describe("lifecycle methods", () => {
    it("componentDidMount with `params` from the URL", () => {
      const component = shallow(<Dashboard {...dispatchProps} {...reducerProps} />);
      const instance = component.instance();
      const renderDataFromURLSpy = jest.spyOn(instance, "renderDataFromURL");
      reducerProps.location.search = "?page=1&per_page=25";
      const params = reducerProps.location.search;
      instance.renderDataFromURL(params);
      expect(renderDataFromURLSpy).toHaveBeenCalledWith(params);
      expect(dispatchProps.fetchStatusOptions).toHaveBeenCalled();
      expect(dispatchProps.fetchRegionOptions).toHaveBeenCalled();
    });

    it("componentDidMount without `params` from the URL", () => {
      const component = shallow(<Dashboard {...dispatchProps} {...reducerProps} />);
      component.update();
      reducerProps.history.push(
        router.MINE_HOME_PAGE.dynamicRoute({
          page: String.DEFAULT_PAGE,
          per_page: String.DEFAULT_PER_PAGE,
        })
      );
      expect(reducerProps.history.push).toHaveBeenCalledWith(
        router.MINE_HOME_PAGE.dynamicRoute({
          page: String.DEFAULT_PAGE,
          per_page: String.DEFAULT_PER_PAGE,
        })
      );
      expect(dispatchProps.fetchStatusOptions).toHaveBeenCalled();
      expect(dispatchProps.fetchRegionOptions).toHaveBeenCalled();
    });
  });
});
