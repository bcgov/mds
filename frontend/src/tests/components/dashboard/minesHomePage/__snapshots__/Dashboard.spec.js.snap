// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dashboard
    closeModal={[MockFunction]}
    createMineRecord={[MockFunction]}
    fetchApplicationStatusOptions={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    fetchMineCommodityOptions={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    fetchMineDisturbanceOptions={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    fetchMineRecords={
      [MockFunction] {
        "calls": Array [
          Array [
            " ",
          ],
        ],
      }
    }
    fetchMineTenureTypes={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    fetchPartyRelationshipTypes={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    fetchPermitStatusOptions={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    fetchRegionOptions={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    fetchStatusOptions={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    history={
      Object {
        "location": Object {},
        "push": [MockFunction],
      }
    }
    location={
      Object {
        "search": " ",
      }
    }
    mineDisturbanceOptions={
      Object {
        "options": Array [
          Object {
            "description": "Surface",
            "exclusive_ind": false,
            "mine_disturbance_code": "SUR",
            "mine_tenure_type_codes": Array [
              "COL",
              "MIN",
              "PLR",
              "BCL",
            ],
          },
          Object {
            "description": "Underground",
            "exclusive_ind": false,
            "mine_disturbance_code": "UND",
            "mine_tenure_type_codes": Array [
              "COL",
              "MIN",
              "PLR",
            ],
          },
          Object {
            "description": "Coal Wash",
            "exclusive_ind": true,
            "mine_disturbance_code": "CWA",
            "mine_tenure_type_codes": Array [
              "COL",
            ],
          },
          Object {
            "description": "Mill",
            "exclusive_ind": true,
            "mine_disturbance_code": "MIL",
            "mine_tenure_type_codes": Array [
              "PLR",
            ],
          },
        ],
      }
    }
    mineIds={
      Array [
        "18133c75-49ad-4101-85f3-a43e35ae989a",
        "18145c75-49ad-0101-85f3-a43e45ae989a",
      ]
    }
    mineRegionHash={
      Object {
        "NE": "North East",
        "NW": "North West",
        "SC": "South Central",
        "SE": "South East",
        "SW": "South West",
      }
    }
    mineRegionOptions={
      Array [
        Object {
          "label": "South West",
          "value": "SW",
        },
        Object {
          "label": "South Central",
          "value": "SC",
        },
        Object {
          "label": "North West",
          "value": "NW",
        },
        Object {
          "label": "North East",
          "value": "NE",
        },
        Object {
          "label": "South East",
          "value": "SE",
        },
      ]
    }
    mineStatusOptions={
      Array [
        Object {
          "label": "Abandoned",
          "value": "ABN",
        },
        Object {
          "children": Array [
            Object {
              "label": "Care & Maintenance",
              "value": "CM",
            },
            Object {
              "children": Array [
                Object {
                  "label": "Long Term Maintenance",
                  "value": "LTM",
                },
                Object {
                  "label": "Long Term Maintenance Water Treatment",
                  "value": "LWT",
                },
                Object {
                  "label": "Permit Release Pending",
                  "value": "PRP",
                },
              ],
              "label": "Reclamation",
              "value": "REC",
            },
            Object {
              "children": Array [
                Object {
                  "label": "Long Term Maintenance",
                  "value": "LTM",
                },
                Object {
                  "label": "Long Term Maintenance Water Treatment",
                  "value": "LWT",
                },
                Object {
                  "label": "Reclamation Not Starte",
                  "value": "RNS",
                },
                Object {
                  "label": "Site Visit Required",
                  "value": "SVR",
                },
              ],
              "label": "Orphaned",
              "value": "ORP",
            },
            Object {
              "label": "Unknown",
              "value": "UN",
            },
          ],
          "label": "Closed",
          "value": "CLD",
        },
        Object {
          "label": "Not Started",
          "value": "NS",
        },
        Object {
          "children": Array [
            Object {
              "label": "Year round",
              "value": "YR",
            },
            Object {
              "label": "Seasonal",
              "value": "SEA",
            },
          ],
          "label": "Operating",
          "value": "OP",
        },
      ]
    }
    mineTenureHash={
      Object {
        "BCL": "BC Land",
        "COL": "Coal",
        "MIN": "Mineral",
        "PLR": "Placer",
      }
    }
    mineTenureTypes={
      Array [
        Object {
          "label": "Coal",
          "value": "COL",
        },
        Object {
          "label": "Mineral",
          "value": "MIN",
        },
        Object {
          "label": "Placer",
          "value": "PLR",
        },
        Object {
          "label": "BC Land",
          "value": "BCL",
        },
      ]
    }
    mines={
      Object {
        "18133c75-49ad-4101-85f3-a43e35ae989a": Object {
          "guid": "18133c75-49ad-4101-85f3-a43e35ae989a",
          "mine_expected_documents": Array [
            Object {
              "date_created": "2018-11-19",
              "due_date": "",
              "exp_document_guid": "806608ed-d2b4-4f83-8b22-739d5223c56f",
              "exp_document_name": "OMS Manual",
              "exp_document_status": Object {
                "description": "Not Received",
                "exp_document_status_code": "MIA",
              },
              "mine_guid": "18133c75-49ad-4101-85f3-a43e35ae989a",
              "related_documents": Array [],
              "req_document_guid": "c95886bc-e4b2-4743-b38d-42eea858e9ee",
            },
          ],
          "mine_location": Object {
            "latitude": null,
            "longitude": null,
          },
          "mine_name": "mine3",
          "mine_no": "BLAH9091",
          "mine_permit": Array [],
          "mine_status": Object {
            "effective_date": "2018-10-16",
            "expiry_date": "9999-12-31",
            "status_labels": Array [
              "Closed",
              "Care & Maintenance",
            ],
            "statusvalue": Array [
              "CLD",
              "CM",
            ],
          },
          "mine_tailings_storage_facility": Array [
            Object {
              "mine_guid": "18133c75-49ad-4101-85f3-a43e35ae989a",
              "mine_tailings_storage_facility_guid": "e2629897-053e-4218-9299-479375e47f78",
              "mine_tailings_storage_facility_name": "MockTSF",
            },
          ],
          "mine_type": Array [
            Object {
              "mine_tenure_type_code": "PLR",
            },
            Object {
              "mine_tenure_type_code": "MIN",
            },
          ],
          "mineral_tenure_xref": Array [
            Object {
              "tenure_number_id": "1234567",
            },
          ],
          "region_code": "NE",
        },
        "18145c75-49ad-0101-85f3-a43e45ae989a": Object {
          "guid": "18145c75-49ad-0101-85f3-a43e45ae989a",
          "mine_expected_documents": Array [
            Object {
              "date_created": "2018-11-19",
              "due_date": "",
              "exp_document_guid": "806608ed-d2b4-4f83-8b22-739d5223c56f",
              "exp_document_name": "OMS Manual",
              "exp_document_status": Object {
                "description": "Not Received",
                "exp_document_status_code": "MIA",
              },
              "mine_guid": "18145c75-49ad-0101-85f3-a43e45ae989a",
              "related_documents": Array [],
              "req_document_guid": "c95886bc-e4b2-4743-b38d-42eea858e9ee",
            },
          ],
          "mine_location": Object {
            "latitude": null,
            "longitude": null,
          },
          "mine_name": "mine2",
          "mine_no": "BLAH9091",
          "mine_permit": Array [
            Object {
              "authorization_end_date": "9999-12-31",
              "issue_date": "9999-12-31",
              "mine_guid": "1628847c-060b-45f2-990f-815877174801",
              "permit_guid": "2ec66cff-dbf0-4e4b-b9b8-9c7740ba8c81",
              "permit_no": "KNaaWwVdiJ40",
              "permit_status_code": "Z",
              "permittee": Array [
                Object {
                  "effective_date": "2018-10-02",
                  "expiry_date": "9999-12-31",
                  "party": Object {
                    "effective_date": "2018-10-03",
                    "email": "JgHxeyjv@aezZIwee.com",
                    "expiry_date": "9999-12-31",
                    "first_name": "Tiyudoveh",
                    "name": "Tiyudoveh Higesewawa",
                    "party_guid": "1c7da2c4-10d5-4c9f-994a-96427aa0c69b",
                    "party_name": "Higesewawa",
                    "party_type_code": "PER",
                    "phone_ext": null,
                    "phone_no": "123-123-1234",
                  },
                  "party_guid": "1c7da2c4-10d5-4c9f-994a-96427aa0c69b",
                  "party_name": "Yivihoke",
                  "permit_guid": "2ec66cff-dbf0-4e4b-b9b8-9c7740ba8c81",
                  "permittee_guid": "abd4c8bb-48ac-4ec2-b296-5f38d4118176",
                },
              ],
            },
            Object {
              "authorization_end_date": "9999-12-31",
              "issue_date": "9999-12-31",
              "mine_guid": "1628847c-060b-45f2-990f-815877174801",
              "permit_guid": "1877097f-bf9d-40c8-a5fa-53d9a79623c9",
              "permit_no": "xfM0c0ZKEw7B",
              "permit_status_code": "Z",
              "permittee": Array [
                Object {
                  "effective_date": "2018-10-02",
                  "expiry_date": "9999-12-31",
                  "party": Object {
                    "effective_date": "2018-10-03",
                    "email": "JgHxeyjv@aezZIwee.com",
                    "expiry_date": "9999-12-31",
                    "first_name": "Tiyudoveh",
                    "name": "Tiyudoveh Higesewawa",
                    "party_guid": "1c7da2c4-10d5-4c9f-994a-96427aa0c69b",
                    "party_name": "Higesewawa",
                    "party_type_code": "PER",
                    "phone_ext": null,
                    "phone_no": "123-123-1234",
                  },
                  "party_guid": "1c7da2c4-10d5-4c9f-994a-96427aa0c69b",
                  "party_name": "Yivihoke",
                  "permit_guid": "1877097f-bf9d-40c8-a5fa-53d9a79623c9",
                  "permittee_guid": "3491c9a5-8f09-471f-bb1b-3ea246eb9796",
                },
              ],
            },
          ],
          "mine_status": Object {
            "effective_date": "2018-10-16",
            "expiry_date": "9999-12-31",
            "status_labels": Array [
              "Closed",
              "Care & Maintenance",
            ],
            "statusvalue": Array [
              "CLD",
              "CM",
            ],
          },
          "mine_tailings_storage_facility": Array [
            Object {
              "mine_guid": "18145c75-49ad-0101-85f3-a43e45ae989a",
              "mine_tailings_storage_facility_guid": "e2629897-053e-4218-9299-479375e47f34",
              "mine_tailings_storage_facility_name": "MockTSF1",
            },
          ],
          "mine_type": Array [
            Object {
              "mine_tenure_type_code": "PLR",
            },
            Object {
              "mine_tenure_type_code": "MIN",
            },
          ],
          "mineral_tenure_xref": Array [
            Object {
              "tenure_number_id": "1234567",
            },
          ],
          "region_code": "NE",
        },
      }
    }
    openModal={[MockFunction]}
    pageData={
      Object {
        "current_page": 1,
        "has_next": true,
        "has_prev": false,
        "items_per_page": 25,
        "mines": Array [],
        "next_num": 2,
        "prev_num": null,
        "total": 9000,
        "total_pages": 360,
      }
    }
    setOptionsLoaded={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="landing-page__header"
        >
          <div
            className="inline-flex between center-mobile center-mobile"
          >
            <div>
              <h1>
                Mine Lookup
              </h1>
              <p>
                To find a mine summary, search in the list or map section below.
              </p>
            </div>
            <div>
              <Connect(AuthorizationWrapper)
                permission="role_admin"
              >
                <Button
                  block={false}
                  className="full-mobile"
                  ghost={false}
                  loading={false}
                  onClick={[Function]}
                  prefixCls="ant-btn"
                  type="primary"
                >
                  Create Mine Record
                </Button>
              </Connect(AuthorizationWrapper)>
            </div>
          </div>
        </div>,
        <div
          className="landing-page__content"
        >
          <Loading />
        </div>,
      ],
      "className": "landing-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="inline-flex between center-mobile center-mobile"
          >
            <div>
              <h1>
                Mine Lookup
              </h1>
              <p>
                To find a mine summary, search in the list or map section below.
              </p>
            </div>
            <div>
              <Connect(AuthorizationWrapper)
                permission="role_admin"
              >
                <Button
                  block={false}
                  className="full-mobile"
                  ghost={false}
                  loading={false}
                  onClick={[Function]}
                  prefixCls="ant-btn"
                  type="primary"
                >
                  Create Mine Record
                </Button>
              </Connect(AuthorizationWrapper)>
            </div>
          </div>,
          "className": "landing-page__header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <h1>
                  Mine Lookup
                </h1>
                <p>
                  To find a mine summary, search in the list or map section below.
                </p>
              </div>,
              <div>
                <Connect(AuthorizationWrapper)
                  permission="role_admin"
                >
                  <Button
                    block={false}
                    className="full-mobile"
                    ghost={false}
                    loading={false}
                    onClick={[Function]}
                    prefixCls="ant-btn"
                    type="primary"
                  >
                    Create Mine Record
                  </Button>
                </Connect(AuthorizationWrapper)>
              </div>,
            ],
            "className": "inline-flex between center-mobile center-mobile",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1>
                    Mine Lookup
                  </h1>,
                  <p>
                    To find a mine summary, search in the list or map section below.
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Mine Lookup",
                  },
                  "ref": null,
                  "rendered": "Mine Lookup",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "To find a mine summary, search in the list or map section below.",
                  },
                  "ref": null,
                  "rendered": "To find a mine summary, search in the list or map section below.",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(AuthorizationWrapper)
                  permission="role_admin"
                >
                  <Button
                    block={false}
                    className="full-mobile"
                    ghost={false}
                    loading={false}
                    onClick={[Function]}
                    prefixCls="ant-btn"
                    type="primary"
                  >
                    Create Mine Record
                  </Button>
                </Connect(AuthorizationWrapper)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Button
                    block={false}
                    className="full-mobile"
                    ghost={false}
                    loading={false}
                    onClick={[Function]}
                    prefixCls="ant-btn"
                    type="primary"
                  >
                    Create Mine Record
                  </Button>,
                  "permission": "role_admin",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "block": false,
                    "children": "Create Mine Record",
                    "className": "full-mobile",
                    "ghost": false,
                    "loading": false,
                    "onClick": [Function],
                    "prefixCls": "ant-btn",
                    "type": "primary",
                  },
                  "ref": null,
                  "rendered": "Create Mine Record",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Loading />,
          "className": "landing-page__content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="landing-page__header"
          >
            <div
              className="inline-flex between center-mobile center-mobile"
            >
              <div>
                <h1>
                  Mine Lookup
                </h1>
                <p>
                  To find a mine summary, search in the list or map section below.
                </p>
              </div>
              <div>
                <Connect(AuthorizationWrapper)
                  permission="role_admin"
                >
                  <Button
                    block={false}
                    className="full-mobile"
                    ghost={false}
                    loading={false}
                    onClick={[Function]}
                    prefixCls="ant-btn"
                    type="primary"
                  >
                    Create Mine Record
                  </Button>
                </Connect(AuthorizationWrapper)>
              </div>
            </div>
          </div>,
          <div
            className="landing-page__content"
          >
            <Loading />
          </div>,
        ],
        "className": "landing-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="inline-flex between center-mobile center-mobile"
            >
              <div>
                <h1>
                  Mine Lookup
                </h1>
                <p>
                  To find a mine summary, search in the list or map section below.
                </p>
              </div>
              <div>
                <Connect(AuthorizationWrapper)
                  permission="role_admin"
                >
                  <Button
                    block={false}
                    className="full-mobile"
                    ghost={false}
                    loading={false}
                    onClick={[Function]}
                    prefixCls="ant-btn"
                    type="primary"
                  >
                    Create Mine Record
                  </Button>
                </Connect(AuthorizationWrapper)>
              </div>
            </div>,
            "className": "landing-page__header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <h1>
                    Mine Lookup
                  </h1>
                  <p>
                    To find a mine summary, search in the list or map section below.
                  </p>
                </div>,
                <div>
                  <Connect(AuthorizationWrapper)
                    permission="role_admin"
                  >
                    <Button
                      block={false}
                      className="full-mobile"
                      ghost={false}
                      loading={false}
                      onClick={[Function]}
                      prefixCls="ant-btn"
                      type="primary"
                    >
                      Create Mine Record
                    </Button>
                  </Connect(AuthorizationWrapper)>
                </div>,
              ],
              "className": "inline-flex between center-mobile center-mobile",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h1>
                      Mine Lookup
                    </h1>,
                    <p>
                      To find a mine summary, search in the list or map section below.
                    </p>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Mine Lookup",
                    },
                    "ref": null,
                    "rendered": "Mine Lookup",
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "To find a mine summary, search in the list or map section below.",
                    },
                    "ref": null,
                    "rendered": "To find a mine summary, search in the list or map section below.",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(AuthorizationWrapper)
                    permission="role_admin"
                  >
                    <Button
                      block={false}
                      className="full-mobile"
                      ghost={false}
                      loading={false}
                      onClick={[Function]}
                      prefixCls="ant-btn"
                      type="primary"
                    >
                      Create Mine Record
                    </Button>
                  </Connect(AuthorizationWrapper)>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Button
                      block={false}
                      className="full-mobile"
                      ghost={false}
                      loading={false}
                      onClick={[Function]}
                      prefixCls="ant-btn"
                      type="primary"
                    >
                      Create Mine Record
                    </Button>,
                    "permission": "role_admin",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "block": false,
                      "children": "Create Mine Record",
                      "className": "full-mobile",
                      "ghost": false,
                      "loading": false,
                      "onClick": [Function],
                      "prefixCls": "ant-btn",
                      "type": "primary",
                    },
                    "ref": null,
                    "rendered": "Create Mine Record",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Loading />,
            "className": "landing-page__content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
